packages/std/src/math/decimal.rs

```diff
@@ -609,6 +636,31 @@ mod tests {
         dec /= Uint128(0);
     }
 
+    #[test]
+    fn decimal_uint128_sqrt() {
+        assert_eq!(Decimal::percent(900).sqrt(), Decimal::percent(300));
+
+        assert!(Decimal::percent(316) < Decimal::percent(1000).sqrt());
+        assert!(Decimal::percent(1000).sqrt() < Decimal::percent(317));
+    }
+
+    /// sqrt(2) is an irrational number, i.e. all 18 decimal places should be used.
+    #[test]
+    fn decimal_uint128_sqrt_is_precise() {
+        assert_eq!(
+            Decimal::from_str("2").unwrap().sqrt(),
+            Decimal::from_str("1.414213562373095048").unwrap() // https://www.wolframalpha.com/input/?i=sqrt%282%29
+        );
+    }
```

Comment:
Can we add a test where one of the other `i` values < 9 is used? Then we should see traiing errors.

Reply:
Yep!